src/sentry/integrations/slack/actions/notification.py

```diff
@@ -120,12 +129,71 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
                     "attachments": json.dumps(attachments),
                 }
 
+            rule = rules[0] if rules else None
+            rule_to_use = self.rule if self.rule else rule
```

Comment:
Is there a way to simplify this? what is rules vs self.rule

Reply:
rules is something that is passed in through the future pipeline, and self.rule is the instance attribute that was added in through the initializer. I do not know why we pass in rules through the future :/